response_test.go

```diff
@@ -987,6 +988,87 @@ func TestNullableAttr_Bool(t *testing.T) {
 	}
 }
 
+func TestNullableRelationship(t *testing.T) {
+	comment := &Comment{
+		ID:   5,
+		Body: "Hello World",
+	}
+
+	for _, tc := range []struct {
+		desc         string
+		input        *WithNullableAttrs
+		verification func(data map[string]interface{}) error
+	}{
+		{
+			desc: "nullable_comment_unspecified",
+			input: &WithNullableAttrs{
+				ID:              5,
+				NullableComment: nil,
+			},
+			verification: func(root map[string]interface{}) error {
+				_, ok := root["data"].(map[string]interface{})["relationships"]
+
+				if got, want := ok, false; got != want {
+					return fmt.Errorf("got %v, want %v", got, want)
+				}
+				return nil
+			},
+		},
+		{
+			desc: "nullable_comment_null",
+			input: &WithNullableAttrs{
+				ID:              5,
+				NullableComment: NewNullNullableRelationship[*Comment](),
+			},
+			verification: func(root map[string]interface{}) error {
+				commentData, ok := root["data"].(map[string]interface{})["relationships"].(map[string]interface{})["nullable_comment"].(map[string]interface{})["data"]
+
+				if got, want := ok, true; got != want {
+					return fmt.Errorf("got %v, want %v", got, want)
+				}
+
+				if commentData != nil {
+					return fmt.Errorf("Expected nil data for nullable_comment but was '%v'", commentData)
+				}
+				return nil
+			},
+		},
+		{
+			desc: "nullable_comment_not_null",
```

Comment:
If we were to want to support de-serializing slice types we would implement tests for that here. 
As it's not included in the PR those sorts of tests are omitted here.